src/components/inputs/localized-multiline-text-input/localized-multiline-text-input.js

```diff
@@ -21,253 +22,244 @@ import LanguagesControl from './languages-control';
 // multiline inputs, while we use show/hide/open/close for the remaining
 // languages.
-export class LocalizedMultilineTextInput extends React.Component {
-  static displayName = 'LocalizedMultilineTextInput';
-
-  static RequiredValueErrorMessage = RequiredValueErrorMessage;
+const LocalizedMultilineTextInput = props => {
+  const intl = useIntl();
 
-  static propTypes = {
-    id: PropTypes.string,
-    name: PropTypes.string,
-    autoComplete: PropTypes.string,
-    // then input doesn't accept a "languages" prop, instead all possible
-    // languages have to exist (with empty or filled strings) on the value:
-    //   { en: 'foo', de: '', es: '' }
-    value: PropTypes.objectOf(PropTypes.string).isRequired,
-    onChange: requiredIf(PropTypes.func, props => !props.isReadOnly),
-    selectedLanguage: PropTypes.string.isRequired,
-    onBlur: PropTypes.func,
-    onFocus: PropTypes.func,
-    defaultExpandMultilineText: PropTypes.bool,
-    hideLanguageExpansionControls: PropTypes.bool,
-    defaultExpandLanguages: (props, propName, componentName, ...rest) => {
-      if (
-        props.hideLanguageExpansionControls &&
-        typeof props[propName] === 'boolean'
-      ) {
-        throw new Error(
-          oneLine`
-            ${componentName}: "${propName}" does not have any effect when
-            "hideLanguageExpansionControls" is set.
-          `
-        );
-      }
-      return PropTypes.bool(props, propName, componentName, ...rest);
+  const initialExpandedTranslationsState = Object.keys(props.value).reduce(
+    (translations, locale) => {
+      return {
+        [locale]: Boolean(props.defaultExpandMultilineText),
+        ...translations,
+      };
     },
-    isAutofocussed: PropTypes.bool,
-    isDisabled: PropTypes.bool,
-    isReadOnly: PropTypes.bool,
-    placeholder: PropTypes.objectOf(PropTypes.string),
-    horizontalConstraint: PropTypes.oneOf(['m', 'l', 'xl', 'scale']),
-    hasError: PropTypes.bool,
-    hasWarning: PropTypes.bool,
-    errors: PropTypes.objectOf(PropTypes.node),
-    warnings: PropTypes.objectOf(PropTypes.node),
-    // HoC
-    intl: PropTypes.shape({
-      formatMessage: PropTypes.func.isRequired,
-    }).isRequired,
-  };
+    {}
+  );
+
+  const [
+    expandedTranslationsState,
+    expandedTranslationsDispatch,
+  ] = React.useReducer(
+    expandedTranslationsReducer,
+    initialExpandedTranslationsState
+  );
 
-  static getId = getId;
+  const defaultExpansionState =
+    props.hideLanguageExpansionControls ||
+    props.defaultExpandLanguages ||
+    // useToggleState's default is `true`, but we want `false`
+    false;
 
-  static getName = getName;
+  const [areLanguagesOpened, toggleLanguages] = useToggleState(
+    defaultExpansionState
+  );
 
-  static defaultProps = {
-    horizontalConstraint: 'scale',
+  const toggleLanguage = language => {
+    expandedTranslationsDispatch({ type: 'toggle', language });
```

Comment:
Nit: if we model the actions a-la redux, we should use `{ type, payload }`

```suggestion
    expandedTranslationsDispatch({ type: 'toggle', payload: language });
```